isaac/controllers.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Controller Tags and Cross-Chain Resources
+category: distributed-systems
+co_authors: 
+publish_date: 
+image: media/
+imageAlt: 
+imageCaption: 
+excerpt: 
+---
+
+Here we set out to _generalize_ the "wrapped tokens" from [IBC](https://github.com/cosmos/ibc/blob/8d9d3b6fe7309b034df8457760b3bbd11d24b8e1/archive/papers/2020-05/build/paper.pdf#L2) and [ICS20](https://github.com/cosmos/ibc/blob/main/spec/app/ics-020-fungible-token-transfer/README.md).
+We create a sound and efficient way to safely transfer arbitrary mutable digital objects (not just tokens) between mutually distrusting chains, tracking precisely how "trustworthy" each object is, and what "promises" each chain has made. 
+Our approach is based on [taint tracking](http://www.tcse.cn/~wsdou/papers/2022-dsn-dista.pdf) and _endorsement_ from [Information Flow Control](https://en.wikipedia.org/wiki/Information_flow_(information_theory)) research, with the added twist that modern blockchains can have the power to _review_ and _verify_ histories of other chains, using [recursive ZKPs](https://eprint.iacr.org/2021/370), [optimistic proofs](https://www.coindesk.com/tech/2024/03/19/optimism-finally-starts-testing-fault-proofs-at-heart-of-design-and-of-criticism/), or simply fully replicating the other chain's state machine.
+
+TODO: actual ART-report link
```

Comment:
This exists now - https://doi.org/10.5281/zenodo.10498997

Reply:
Or if you wanted the art.anoma.net link - https://art.anoma.net/list.html#paper-10498997